staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go

```diff
@@ -259,6 +259,9 @@ func (s *Validator) validateExpressions(ctx context.Context, fldPath *field.Path
 			continue
 		}
 		if evalResult != types.True {
+			if len(rule.FieldPath) != 0 {
+				fldPath = fldPath.Child(sts.XValidations[i].FieldPath[1:])
```

Comment:
Is this right? Does adding a JSON path string (potentially with multiple levels of depth) as a child to a field path create a valid field path?  Are there differences between JSON path and field paths (e.g. tolerance for extra whitespace?)

Reply:
Update to use validateFieldPath to return a valid path and added tests